TeleportMenu.lua

```diff
@@ -516,15 +497,19 @@ end
 function tpm:updateMageFlyouts()
 	local teleportButton = TeleportMeButtonsFrame.mageTeleportButton
 	local portalButton = TeleportMeButtonsFrame.magePortalButton
+	if not teleportButton or not portalButton then
+		return
+	end
 	local _, _, _, _, teleportYOffset = teleportButton:GetPoint()
 	local _, _, _, _, portalYOffset = portalButton:GetPoint()
 
-	if select(4, GetFlyoutInfo(12)) then -- Player is Alliance
-		local updatedTeleportButton = tpm:CreateFlyout(1, 237509, teleportYOffset)
-		local updatedPortalButton = tpm:CreateFlyout(11, 135748, portalYOffset)
-	else -- Player is Horde
-		local updatedTeleportButton = tpm:CreateFlyout(8, 237509, teleportYOffset)
-		local updatedPortalButton = tpm:CreateFlyout(12, 135748, portalYOffset)
+	local updatedTeleportButton, updatedPortalButton
+	if select(4, GetFlyoutInfo(1)) then -- Player is Horde
+		updatedTeleportButton = tpm:CreateFlyout(tpTable[7], teleportYOffset)
+		updatedPortalButton = tpm:CreateFlyout(tpTable[9], portalYOffset)
+	else -- Player is Alliance
+		updatedTeleportButton = tpm:CreateFlyout(tpTable[8], teleportYOffset)
+		updatedPortalButton = tpm:CreateFlyout(tpTable[10], portalYOffset)
```

Comment:
Maybe it is easier to simply access the flyout buttons directly and reverse them if possible :thinking:

Reply:
I changed the code to simply work with xOffset instead of relying on creating the buttons and reversing the for loop. At least to me this is much simpler approach and works well.